### DISCUSSION TITLE
The repository 'https://xpra.org/beta jammy InRelease' is not signed

### DISCUSSION THREAD

**Discussion by mmol67** [Xpra-org/xpra#3892] [Q&A] [answered]
For stable and beta, various ubuntu releases

I've read this https://github.com/Xpra-org/xpra/issues/3846

but the solution

```
tmp_keyring=$(mktemp)
keyid=$(gpg --show-key --with-colons xpra.asc | grep ^fpr: | head -n 1 | cut -d: -f10)
gpg --primary-keyring $tmp_keyring --recv-keys $keyid
gpg --primary-keyring $tmp_keyring --export -a $keyid > xpra.asc
rm $tmp_keyring
```
is not working for me. What to do?

**Comment by totaam** [comment]
Until a new release lands in the stable repositories, you can try the beta repositories which are signed using the new key `0x73254CAD17978FAF`, more info here: #3863

**Comment by mmol67** [comment]
I'm trying with last gpg.asc and xpra-2023.gpg (changing in in sources.list) and It does not work. I'll wait for the new download instructions and when all this is fixed. Thank you very much.

**Comment by totaam** [comment]
@friki67 which repository?  what exact error message?

**Reply by mmol67** [reply]
I'm not at home right now, so I cannot check it once more. What I used was  Jammy beta repository and xpra-2023.gpg
The error was "repository not signed".

**Reply by mmol67** [reply]
What I do:
1)` sudo wget -O "/usr/share/keyrings/xpra-2023.gpg" https://xpra.org/xpra-2023.gpg`
2) as root
```
cd /etc/apt/sources.list.d
wget https://raw.githubusercontent.com/Xpra-org/xpra/master/packaging/repos/jammy/xpra-beta.sources
```

3) Edit this file and change as this:

                                            
```
Types: deb
URIs: https://xpra.org/beta
Suites: jammy
Components: main
Signed-By: /usr/share/keyrings/xpra-2023.gpg
Architectures: amd64 arm64
```
apt update
```
W: GPG error: https://xpra.org/beta jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 73254CAD17978FAF
E: The repository 'https://xpra.org/beta jammy InRelease' is not signed.
```

I've tried too using gpg.asc from the other place, with same result. Ubuntu is not my thing so maybe I'm messing something.

**Reply by totaam** [reply, marked as answer]
> `NO_PUBKEY 73254CAD17978FAF`

I eventually hit the same issue as you... and the solution is a bit of magic.
The file had the wrong extension: it was an armored version of the key and Debian relies on the `.asc` file extension to load it.

So I have uploaded the binary version in the right place, and now we have:
* binary key https://xpra.org/xpra-2023.gpg
* armored key https://xpra.org/xpra-2023.asc
Use whichever one you prefer, but don't rename the file!

**Reply by mmol67** [reply]
Fixed. Thank you very much!
